obsproject/obs-studio
# UI: Don't calculate physical screen size for Wayland

### Description
The UI code assumes QScreen.size is in the logical screen resolution space. This appears to be true with X11 (and probably Win32/MacOS as well), but not with Wayland sessions. Qscreen.size is the physical screen resolution on Wayland.

This PR skips physical resolution conversion when on a Wayland backend.

### Motivation and Context
Currently, the screen resolution found in the settings UI will **not** match the actual resolution on Wayland if DPI scaling is used.

![imagem](https://github.com/user-attachments/assets/b88d9567-3624-434e-ac26-79269f24d22a)

This PR fixes this issue, making the Canvas resolution match the user's actual screen resolution in the settings UI and auto-configuration wizard:

![imagem](https://github.com/user-attachments/assets/10da3f2a-1d0e-4686-b5c5-cda27ce359f6)

### How Has This Been Tested?

Environment:
- OS: Fedora Workstation 41 (Rawhide)
- Machine: Thinkpad P1 Gen 6
  - CPU: i7-13800h
  - Graphics: Intel Gen12 Graphics (Intel Xe igpu) + Nvidia RTX 4000 Ada (Laptop) [Nvidia 560 open]
  - Desktop: GNOME 47 (Alpha)
  - Display: 3840x2400 (16:10)

I've tested the Wayland and X11 backends. On X11, there's no changes with the screen resolution option (which was already correct). On Wayland, the screen resolution now matches the real resolution.

Steps: 
1. Set the screen's DPI to any value other than 1.
2. Launch OBS studio (in Wayland)
3. Check the "Output" tab in the Settings UI - Ensure the first Canvas resolution in the list is equal to the screen's resolution.
4. Same step as above, but in the Auto-configuration wizard.

Since this only adds a single conditional check for Wayland, this change should be isolated to Linux Desktop Wayland targets. X11, Win32 and MacOS should not be affected. 

### Types of changes

- Bug fix (non-breaking change which fixes an issue)







### Checklist:
